src/Illuminate/Support/Collection.php

```diff
@@ -1079,6 +1079,10 @@ public function put($key, $value)
      */
     public function random($amount = null)
     {
+        if ($amount === 0) {
+            return new static;
+        }
+
         if (($requested = $amount ?: 1) > ($count = $this->count())) {
```

Comment:
Now that `0` is a supported value, this `?:` looks a bit dangerous. With PHP 7 we could just replace it with `??`, but here with PHP 5 support, I'd suggest the following:

```php
$requested = is_null($amount) ? 1 : $amount;
$count = $this->count();

if ($requested > $count) {
    throw new InvalidArgumentException(
        "You requested {$requested} items, but there are only {$count} items in the collection."
    );
}
```